lib/exercises.rb

```diff
@@ -1,19 +1,44 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2) - Nested loop
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  anagram_hash = {}
+
+  strings.each do |str|
+    bucket = anagram_hash.keys.find do |b|
+      unique_chars = b.chars.sort.join
+      unique_chars == str.chars.sort.join
+    end
```

Comment:
Since you're using a hash you don't need to do a `find` here.

Instead consider:

```ruby
letters = str.chars.sort
if anagram_hash[letters]
  anagram_hash[letters] << str
else
  anagrams_hash[letters] = [str]
end
```